### PR TITLE
Opal 305

### PR DESCRIPTION
**Pull request by BE-USAF**

### CODE REVIEW

File: .github/workflows/pytest_and_ruff.yml [hunk 1]
```diff
@@ -108,7 +96,7 @@ jobs:
         python3.10 -m pip install pytest
     # If you do not install Weave's dependencies then it will fail PyLint due to import-errors
     - name: Run pylint
-      run: pylint weave weave/tests
```
Comment (AB-USAF) [inline .github/workflows/pytest_and_ruff.yml:111]: Why did you change pylint to only test that specific test? It should run on all tests

File: weave/index/index_sql.py [hunk 2]
```diff
@@ -59,22 +51,29 @@ def __init__(self, file_system, pantry_path, **kwargs):
...
             # Pylint thinks this is pointless, so pylint is getting ignored.
             # pylint: disable=pointless-statement
-            os.environ["MSSQL_HOST"]
-            os.environ["MSSQL_USERNAME"]
-            os.environ["MSSQL_PASSWORD"]
+            self._sql_connection = {
+                'host': os.environ["WEAVE_SQL_HOST"],
+                'username': os.environ["WEAVE_SQL_USERNAME"],
+                'password': os.environ["WEAVE_SQL_PASSWORD"],
+                'port': os.environ.get("WEAVE_SQL_PORT", 5432),
```
Comment (AB-USAF) [inline weave/index/index_sql.py:58]: is the port 5432 a standard or something or just a random number?

File: weave/index/index_sql.py [hunk 3]
```diff
@@ -59,22 +51,29 @@ def __init__(self, file_system, pantry_path, **kwargs):
...
+                          ) from key_error
 
         self._file_system = file_system
         self._pantry_path = pantry_path
 
         # Set the database name (defaults to weave_db). DATABASE MUST ALREADY
         # EXIST. If it does not, the user must create it manually.
-        self._database_name = kwargs.get("database_name", "weave_db")\
-            .replace("-", "_")
+        self._database_name = kwargs.get(
```
Comment (AB-USAF) [inline weave/index/index_sql.py:72]: I know you didn't write this comment, but what happens if the database doesn't already exist?

File: weave/index/index_sql.py [hunk 4]
```diff
@@ -273,81 +256,14 @@ def generate_index(self, **kwargs):
...
-                        commit=True,
-                    )
-
-        if len(bad_baskets) != 0:
-            warnings.warn("baskets found in the following locations "
-                          "do not follow specified weave schema:\n"
-                          f"{bad_baskets}")
+            entry = create_index_from_fs(basket_json_address,
+                                         file_system=self.file_system)
+            self.track_basket(entry)
```
Comment (AB-USAF) [inline weave/index/index_sql.py:266]: I still love this so much :goat:

File: weave/index/index_sql.py [hunk 5]
```diff
@@ -586,20 +497,20 @@ def get_parents(self, basket_address, **kwargs):
...
-                    CAST(parent_uuids.parent_uuid AS nvarchar(max)) AS id,
-                    CAST(child_record.path + '/' + parent_uuids.parent_uuid
-                        AS nvarchar(max)) AS path
+                    CAST(parent_uuids.parent_uuid AS varchar) AS id,
+                    CAST(child_record.path || '/' || parent_uuids.parent_uuid
+                        AS varchar) AS path
                 FROM {self.pantry_schema}.parent_uuids
                 JOIN child_record ON parent_uuids.uuid = child_record.id
-                WHERE 
+                WHERE
```
Comment (AB-USAF) [inline weave/index/index_sql.py:513]: I'm glad this was fairly easy to convert

**Review by AB-USAF — CHANGES_REQUESTED** [review]
I need clarification or change on the pylint workflow

File: .github/workflows/pytest_and_ruff.yml (hunk 1 above)
Comment (BE-USAF) [reply .github/workflows/pytest_and_ruff.yml:111]: Great catch... I did that to speed up my testing iteration. That would've been bad to keep in. Thank you!

File: weave/index/index_sql.py (hunk 2 above)
Comment (BE-USAF) [reply weave/index/index_sql.py:58]: 5432 is the Postgres default port. I thought it would be a good idea to keep it because it will very often be true.

File: weave/index/index_sql.py (hunk 3 above)
Comment (BE-USAF) [reply weave/index/index_sql.py:72]: If it doesn't exist, we get a hard failure from the sqla.create_engine call on line 86. This is not something that I think we can (or should) create in the python, we just want to connect to something that exists. Typically, you need to create the database outside of Python as an admin.

**Review by AB-USAF — APPROVED** [review]

**Review by MJ-USAF — APPROVED** [review]
Nice work. That looked like a total bear having to wait for github to run all the pytests.
